tests/worker/test_replayer.py

```diff
@@ -260,130 +247,65 @@ async def test_replayer_workflow_not_registered(client: Client) -> None:
         await handle.result()
 
     # Collect history and replay it expecting error
-    history = await get_history(client, handle.id)
     with pytest.raises(RuntimeError) as err:
-        await Replayer(workflows=[SayHelloWorkflowDifferent]).replay_workflow(history)
+        await Replayer(workflows=[SayHelloWorkflowDifferent]).replay_workflow(
+            await handle.fetch_history()
+        )
     assert "SayHelloWorkflow is not registered" in str(err.value)
 
 
+async def test_replayer_multiple_from_client(
+    client: Client, env: WorkflowEnvironment
+) -> None:
+    if env.supports_time_skipping:
+        pytest.skip("Java test server doesn't support newer workflow listing")
+
+    # Run 5 say-hello's, with second and 4th having non-det errors. Use the same
+    # workflow ID for all of them so that we can query.
+    workflow_id_suffix = f"-{uuid.uuid4()}"
+    async with new_say_hello_worker(client) as worker:
+        workflow_ids = []
+        for i in range(5):
+            workflow_id = f"workflow-{i}{workflow_id_suffix}"
+            workflow_ids.append(workflow_id)
+            await client.execute_workflow(
+                SayHelloWorkflow.run,
+                SayHelloParams(
+                    name="Temporal", should_cause_nondeterminism=i == 1 or i == 3
+                ),
+                id=workflow_id,
+                task_queue=worker.task_queue,
+            )
+
+    # Lazily list then fetch histories the workflows, manually filtering for
+    # the ones we expect
+    async def hist_iter() -> AsyncIterator[WorkflowHistory]:
```

Comment:
Can we show one as well that doesn't use an intermediate user-created iterator? Nice to just show the pieces end-to-end that the user would typically use.

Reply:
Yes. I had that originally, but standard visibility doesn't support `or` or `in` clauses and I have to be careful to remain isolated in this test (none of my tests step on each other).

The only way to do this right would be to reuse a workflow ID across many runs so I can have `WorkflowId = 'foo'` as the query. I can't find any other standard-visibility supported attributes. I didn't do this because I was lazy and wanted different workflow IDs so I could ensure list order (standard visibility doesn't support ordering either). But I can just check by run ID.

Reply:
Updated test to share workflow IDs across runs